docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,763 @@
+              on ad.alarm_definition_id = probable_causes.alarm_definition_id
+WHERE probable_causes.probable_cause_id = 'f5ac4ac7-0ff3-40a2-b305-77313c28136a';
+```
+
+## K8s resources
+We will need few K8s resources that will be eventually applied by the Operator. 
+
+#### Alarm server
+This is essentially a typical CRUD app and we need the following
+
+- Deployment: It should have one initContainers that runs to completion to perform DB migration and one main container which starts the main server. 
+  No HA, so set replica to 1. It should also contain all the ENV variables needed to talk to postgres deployment DB_HOST, DB_PORT, DB_NAME, DB_USER etc.
+  Suitable resources should be provided but not much memory and CPU is need for CRUD apps. 
+- Secrets: DB creds and configs should be read from postgres deployment. 
+- Service: Expose and balance using `ClusterIP` (though to start with we will set replica to 1)
+- Ingress: Expose service so that it can be called by users from outside the cluster
+
+
+#### Postgres
+This deployment can be leveraged by many microservices by creating their own Database. 
+
+- Deployment: One replica should be good enough for our case. 
+  Default username, password and dbname (note this is simply to allow postgres pod initialize) should be provided. 
+  Deployment should also mount to the right path `/var/lib/postgresql/data` (please check the latest docs).
+- PersistentVolumeClaim: At least 20G PVC should be used. Using test, we used about 60k rows DB which took about ~2Gb
+- Service: ClusterIP should be good as it's only used within the cluster. 
+- Secrets: default creds needed to spin postgres, with type: Opaque. This secret will then be ready by 
+   - POSTGRES_USER: `o-ran`
+   - POSTGRES_PASSWORD: `o-ran`
+- ConfigMap: For others to know the hostname and port
+   - POSTGRES_HOST: "postgres.o-ran-namespace.svc.cluster.local"
+   - POSTGRES_PORT: "5432"
+
+With the Secrets and ConfigMap applied, an app may connect to the DB with the following given the service (assuming migration already and `o-ran-infrastructure-monitoring-alarms` already created):
+```go
+connStr := "postgres://o-ran:o-ran@postgres.o-ran-namespace.svc.cluster.local:5432/o-ran-infrastructure-monitoring-alarms?sslmode=disable"
+```
+
+Note:
+- ODF (ceph operator) must be deployed on the hub cluster that database requests a pvc from (same storageclass).
+  This will allow for replicated persistent storage
+
+## Tooling and general dev guidelines
+- Note on concurrency: The new alarm servers provide alarm query, alarm subscription, and alarm notification functional modules and 
+  will be running under the single daemon and single instance of the alarm daemon. 
+  The alarm data entries and alarm subscription entries and other related alarm management 
+  data will be stored in the postgres db tables described in this PR. 
+  1st phase alarm delivery has low scale of alarm subscriptions, and alarm counts to be supported. 
```

Comment:
Not relevant

Reply:
I think we need more detail here.  How is the functionality being partitioned - Query, notifications, subscription handling,  alert manager interface. 

This all be done within a single controller ?

Reply:
Removed the irrelevant text. 

Added the info on functionality. But yeah i think only one server can do the job. It will expose all the external apis (o-ran) and internal apis (alertmanager + future h/w) and handle them concurrently to drive all the features.

Reply:
When you say "handle them concurrently" do you mean through goroutines? I assume this will be necessary, since it won't be OK to block the server while (for example) a notification is being sent to an external subscriber.

Reply:
> through goroutines?

Yeah exactly! The handler starts a new goroutine on every request automatically. So nothing will be blocking. 

As an example, if user calls `/alarms` api and at the alartmanager calls `/caas-alerts` api, it will effectively be two routines. So we can return the `/alarms` call and notify subscribers all at the same time.